source/ncengine/graphics2/diligent/resource/PostProcessColorSinkBufferResource.cpp

```diff
@@ -64,23 +78,46 @@ auto PostProcessColorSinkBufferResource::Add(Diligent::IRenderDevice& device,
             throw NcError("Failed to create texture");
         }
 
-        addedIndices.push_back(static_cast<uint32_t>(m_colorRenderTargets.size()));
-        m_colorRenderTargets.push_back(std::move(pColorRenderTarget));
-        m_colorRenderTargetViewsRT.push_back(m_colorRenderTargets.back()->GetDefaultView(TEXTURE_VIEW_RENDER_TARGET));
-        m_colorRenderTargetViewsSR.push_back(m_colorRenderTargets.back()->GetDefaultView(TEXTURE_VIEW_SHADER_RESOURCE));
+        if (numSamples > 1)
+        {
+            addedIndices.push_back(static_cast<uint32_t>(m_colorRenderTargetsMsaa.size()));
+            m_colorRenderTargetsMsaa.push_back(std::move(pColorRenderTarget));
+            m_colorRenderTargetViewsRTMsaa.push_back(m_colorRenderTargetsMsaa.back()->GetDefaultView(TEXTURE_VIEW_RENDER_TARGET));
+            m_colorRenderTargetViewsSRMsaa.push_back(m_colorRenderTargetsMsaa.back()->GetDefaultView(TEXTURE_VIEW_SHADER_RESOURCE));
+        }
+        else
+        {
+            addedIndices.push_back(static_cast<uint32_t>(m_colorRenderTargets.size()));
+            m_colorRenderTargets.push_back(std::move(pColorRenderTarget));
+            m_colorRenderTargetViewsRT.push_back(m_colorRenderTargets.back()->GetDefaultView(TEXTURE_VIEW_RENDER_TARGET));
+            m_colorRenderTargetViewsSR.push_back(m_colorRenderTargets.back()->GetDefaultView(TEXTURE_VIEW_SHADER_RESOURCE));
+        }
     }
 
-    SetArrayRegion(m_colorRenderTargetViewsSR, 0u, m_colorRenderTargetViewsSR.size());
+    if (numSamples > 1)
+    {
+        SetArrayRegion(m_colorRenderTargetViewsSRMsaa, m_colorRenderTargetViewsSR.size(), m_colorRenderTargetViewsSRMsaa.size());
+    }
+    else
+    {
+        SetArrayRegion(m_colorRenderTargetViewsSR, 0u, m_colorRenderTargetViewsSR.size());
```

Comment:
Didn't look all that closely, but is this end correct? Can this overwrite the first msaa entry?